Core/Services/StitchCoreRemoteMongoDBService/Sources/StitchCoreRemoteMongoDBService/Sync/DataSynchronizer.swift

```diff
@@ -1231,10 +1221,8 @@ public class DataSynchronizer: NetworkStateDelegate, FatalErrorListener {
         var deferredEventEmittingBlock: DeferredBlock? = nil
```

Comment:
[nit] with this new locking scheme, we don't need the deferred event emitting block. Since we're not deferring to unlock the lock, we can just emit the event at the end of the function